### PR TITLE
fix android composer background shadow not visible when image is attached

### LINKED ISSUE #18953: Android composer background shadow not visible when image is attached
### Summary
Android composer background shadow not visible when image is attached

https://github.com/status-im/status-mobile/assets/17097240/2f5a2c17-7c2c-41f0-aff0-b9f07137b3a8

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/18953

### Summary

PR fixes the composer shadow problem, also as bonus makes blur view static instead of animated one.

### Testing notes:
- Please test the composer shadow issue as mentioned in #18953.
- Changes are straightforward and should not break anything, but as the composer is very fragile, we can't say anything for sure. So please let me know if anything is going wrong while opening/closing the compser and sending messages etc.

status: ready

### CODE REVIEW

**Comment by pavloburykh** [comment]
@OmarBasem @alwx hello guys! Could please someone review this PR? Thank you.

**Review by ibrkhalil — APPROVED** [review]

**Review by OmarBasem — APPROVED** [review]

File: src/status_im/contexts/chat/messenger/composer/effects.cljs [hunk 1]
```diff
@@ -70,19 +70,17 @@
...
 (defn empty-effect
-  [{:keys [focused?]}
-   {:keys [container-opacity]}
+  [{:keys [empty-input?]}
    {:keys [input-text images link-previews? reply audio]}]
-  (when (and (not @focused?) (utils/empty-input? input-text images link-previews? reply audio))
-    (reanimated/animate-delay container-opacity constants/empty-opacity 200)))
+  (reanimated/set-shared-value
+   empty-input?
+   (utils/empty-input? input-text images link-previews? reply audio)))
```
Comment (OmarBasem) [inline src/status_im/contexts/chat/messenger/composer/effects.cljs:83]: This was an animation, and now it is setting a shared value. Are you sure about this change @Parveshdhull ?

File: src/status_im/contexts/chat/messenger/composer/effects.cljs [hunk 2]
```diff
@@ -203,12 +198,10 @@
 (defn use-images
   [{:keys [sending-images? input-ref]}
    {:keys [text-value maximized?]}
-   {:keys [container-opacity height saved-height]}
+   {:keys [height saved-height]}
    {:keys [images]}]
   (rn/use-effect
    (fn []
-     (when images
-       (reanimated/animate container-opacity 1))
```
Comment (OmarBasem) [inline src/status_im/contexts/chat/messenger/composer/effects.cljs:211]: Is the container's opacity being animated on adding images somewhere else? (and have you tested adding images, then existing and entering the chat screen)

File: src/status_im/contexts/chat/messenger/composer/gesture.cljs [hunk 3]
```diff
@@ -68,7 +68,6 @@
                             (if-not keyboard-shown
                               (do ; focus and end
                                 (when (< (oops/oget event "velocityY") constants/velocity-threshold)
-                                  (reanimated/set-shared-value container-opacity 1)
```
Comment (OmarBasem) [inline src/status_im/contexts/chat/messenger/composer/gesture.cljs:71]: Also here, is the container's opacity being handled somewhere else on-focus?

File: src/status_im/contexts/chat/messenger/composer/handlers.cljs [hunk 4]
```diff
@@ -164,8 +160,3 @@
       (let [{:keys [start end text-input-handle]} @selection-event]
         (selection/update-selection text-input-handle start end)
         (reset! selection-event nil)))))
-
-(defn layout
-  [event state blur-height]
-  (when (utils/update-blur-height? event state blur-height)
-    (reanimated/set-shared-value blur-height (oops/oget event "nativeEvent.layout.height"))))
```
Comment (OmarBasem) [inline src/status_im/contexts/chat/messenger/composer/handlers.cljs:171]: Do we not need this anymore?

File: src/status_im/contexts/chat/messenger/composer/style.cljs [hunk 5]
```diff
@@ -10,13 +10,12 @@
...
-  [focused? theme]
-  (if platform/ios?
+  [theme]
+  (when platform/ios?
     {:shadow-radius  20
      :shadow-opacity (colors/theme-colors 0.1 0.7 theme)
      :shadow-color   colors/neutral-100
-     :shadow-offset  {:width 0 :height (colors/theme-colors -4 -8 theme)}}
-    {:elevation (if @focused? 10 0)}))
+     :shadow-offset  {:width 0 :height (colors/theme-colors -4 -8 theme)}}))
```
Comment (OmarBasem) [inline src/status_im/contexts/chat/messenger/composer/style.cljs:18]: Is the elevation here not needed? Does the shadow look right with all combinations of being focused and not focused, and having content and not having content?

File: src/status_im/contexts/chat/messenger/composer/style.cljs [hunk 6]
```diff
@@ -95,17 +95,16 @@
     :background-color colors/neutral-95-opa-70}))
 
 (defn blur-container
-  [height focused?]
-  (reanimated/apply-animations-to-style
-   {:height height}
+  [composer-default-height {:keys [blur-container-elevation]}]
+  [{:elevation blur-container-elevation}
    {:position                :absolute
-    :elevation               (if-not @focused? 10 0)
```
Comment (OmarBasem) [inline src/status_im/contexts/chat/messenger/composer/style.cljs:102]: Are we removing the shadow completely on Android?

File: src/status_im/contexts/chat/messenger/composer/effects.cljs (hunk 1 above)
Comment (Parveshdhull) [reply src/status_im/contexts/chat/messenger/composer/effects.cljs:83]: hi @OmarBasem, Thank you very much for reviewing PR.

Yes, the `empty-input?` shared value is supposed to be updated whenever we have input empty without delay. We need animation for container-opacity, which is now handled in `composerContainerOpacity` in the `composer.js` file.

File: src/status_im/contexts/chat/messenger/composer/effects.cljs (hunk 2 above)
Comment (Parveshdhull) [reply src/status_im/contexts/chat/messenger/composer/effects.cljs:211]: Yes, as I mentioned in above comment all animations for container-opacity is handled in composer.js file at one place. Now we don't need to handle this manually.

> and have you tested adding images, then existing and entering the chat screen

Double checked just now, working fine. Thank you for reminding.

File: src/status_im/contexts/chat/messenger/composer/gesture.cljs (hunk 3 above)
Comment (Parveshdhull) [reply src/status_im/contexts/chat/messenger/composer/gesture.cljs:71]: yes

File: src/status_im/contexts/chat/messenger/composer/handlers.cljs (hunk 4 above)
Comment (Parveshdhull) [reply src/status_im/contexts/chat/messenger/composer/handlers.cljs:171]: Composer is only blurred when it is collapsed and empty, not sure why we animated blur height in first place. But yes we don't this anymore

File: src/status_im/contexts/chat/messenger/composer/style.cljs (hunk 5 above)
Comment (Parveshdhull) [reply src/status_im/contexts/chat/messenger/composer/style.cljs:18]: Yes, it looks fine. But just to be double sure, @pavloburykh Please keep an eye on this while testing.

File: src/status_im/contexts/chat/messenger/composer/style.cljs (hunk 6 above)
Comment (Parveshdhull) [reply src/status_im/contexts/chat/messenger/composer/style.cljs:102]: Nope, moved to `blur-container-elevation`

**Comment by pavloburykh** [comment]
@OmarBasem thanks for the review. @Parveshdhull please, ping me up when review comments will be addressed. I have tested the PR and it looks good to me. If there will be any changes based on the review please let me know so I perform re-test. Thank you.

File: src/status_im/contexts/chat/messenger/composer/effects.cljs (hunk 2 above)
Comment (pavloburykh) [reply src/status_im/contexts/chat/messenger/composer/effects.cljs:211]: I also checked this case and it looks fine for me

File: src/status_im/contexts/chat/messenger/composer/style.cljs (hunk 5 above)
Comment (pavloburykh) [reply src/status_im/contexts/chat/messenger/composer/style.cljs:18]: I have checked those cases, shadow looks okay to me

**Comment by pavloburykh** [comment]
@Parveshdhull if not changes are required - PR is ready for merge.

**Pull request merged**